.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,45 @@
+name: Build and deploy
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
based on the `refs/tags` reference at the bottom of this file, should these events include `tags`?

e.g. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-of-specific-tags-occurs

Reply:
The workflow needs to run on all push events to the main branch, so tags and regular commits. So I left it off there and then check later to decide what path to push the artefacts to.

Does that sound right?

Reply:
Hmm. I'm not sure I understand. It looks to me as if this workflow would _never_ run on any `tag` pushes since it's missing something like:

```
on:
  push:
    tags:        
      - "*"
```

Reply:
Ahhh I get it, you mean in addition to the `main` branch. I thought you meant instead of.

Reply:
I've added this now.